Include/internal/pycore_frame.h

```diff
@@ -120,6 +121,15 @@ static inline void _PyFrame_Copy(_PyInterpreterFrame *src, _PyInterpreterFrame *
     // Don't leave a dangling pointer to the old frame when creating generators
     // and coroutines:
     dest->previous = NULL;
+#ifdef Py_GIL_DISABLED
```

Comment:
The top-most frame doesn't necessarily have an up-to-date `stacktop` so you need to scan up to the stack size. 

You could try to keep `stacktop` up to date, but that would require either updating it at every opcode or at every call that might trigger a GC, including every `Py_DECREF`. (I did this in the `nogil` fork, but if you're not careful you can introduce a lot of bugs due to incorrect `stacktop` and it's harder to retrofit efficiently into the current interpreter.)

It's okay for the GC to look at objects that are marked as deferred because only the GC can free them, even if they are above `stacktop`.